build.sh

```diff
@@ -19,8 +24,30 @@ fi
 __buildproj=$__scriptpath/build.proj
 __buildlog=$__scriptpath/msbuild.log
 
-# Run the restore build tools under xbuild to get NuGet and MSBuild pulled down
-xbuild "$__buildproj" /t:_RestoreBuildTools /nologo /verbosity:minimal "/fileloggerparameters:Verbosity=diag;LogFile=$__buildlog"
+# Pull NuGet.exe down if we don't have it already
+if [ ! -e "$__nugetpath" ]; then
+    which curl > /dev/null 2> /dev/null
+    if [ $? -ne 0 ]; then
+        echo "cURL is required to build corefx. Please see https://github.com/dotnet/corefx/wiki/Building-On-Unix for more details."
+        exit 1
+    fi
+    echo "Restoring NuGet.exe..."
+    curl -sSL --create-dirs -o $__nugetpath https://api.nuget.org/downloads/nuget.exe
+    if [ $? -ne 0 ]; then
+        echo "Failed to restore NuGet.exe."
+        exit 1
+    fi
+fi
+
+# Grab the MSBuild package if we don't have it already
+if [ ! -e "$__msbuildpath" ]; then
+    echo "Restoring MSBuild..."
+    mono "$__nugetpath" install $__msbuildpackageid -Version $__msbuildpackageversion -ConfigFile "$__nugetconfig" -OutputDirectory "$__packageroot"
+    if [ $? -ne 0 ]; then
+        echo "Failed to restore MSBuild."
+        exit 1
+    fi
+fi
 
 if [ ! -e "$__msbuildpath" ]; then
```

Comment:
Do we still need this check? If something goes wrong during nuget install of MSBuild in the previous step we should get an error anyway.